### PR TITLE
[REVIEW][OPS] gpuCI Script Updates

### PR DESCRIPTION
**Pull request by mike-wendt**
### Overview
First step before "Project Flash" to improve gpuCI scripts and allow for use of different node types for CPU builds. Post "Project Flash" there will be another round of improvements and updates with the goal to make these scripts easier to maintain.

### Changes
- Consolidates some of the CPU gpuCI scripts, reducing number of scripts to maintain
- Standardizes naming of the upload script to `upload.sh`
- Changes logging to use `gpuci_logger` where practical
- Updates comments and documentation where necessary
- Enables gpuCI to override environment variables based on node types
- Reorders upload script to save ~5 mins or more in testing time

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
rerun tests - merged rapidsai/gpuci-build-environment#132 which contains a fix for using `conda activate rapids` as a non-root user

**Comment by harrism** [comment]
@mike-wendt should this move to 0.17?

**Comment by mike-wendt** [comment]
> @mike-wendt should this move to 0.17?

I've been out, but will work on this today/tomorrow it's almost complete just needs a quick fix and testing so I'd like to get this out 0.16

**Comment by mike-wendt** [comment]
Blocked by the PR rapidsai/gpuci-build-environment#137

**Comment by mike-wendt** [comment]
> Blocked by the PR [rapidsai/gpuci-build-environment#137](https://github.com/rapidsai/gpuci-build-environment/pull/137)

Just merged waiting on new images to build then will retry

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
rerun tests

**Comment by harrism** [comment]
@mike-wendt should this be moved to 0.17?

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
Blocked by rapidsai/gpuci-build-environment#140 and then CUDA 11 tests should pass and we can merge to fix errors seen tonight

**Comment by mike-wendt** [comment]
> Blocked by [rapidsai/gpuci-build-environment#140](https://github.com/rapidsai/gpuci-build-environment/pull/140) and then CUDA 11 tests should pass and we can merge to fix errors seen tonight

This is why I've re-targeted this `branch-0.16` to prevent similar errors occurring during the release

**Comment by mike-wendt** [comment]
rapidsai/gpuci-build-environment#140 merged, waiting for new images and then will re-test

**Comment by mike-wendt** [comment]
Once tests pass will merge in morning

File: ci/gpu/build.sh
Comment (jakirkham) [inline ci/gpu/build.sh:82]:
```suggestion
                  "ucx-proc=*=gpu" \
                  "ucx-py=${MINOR_VERSION}"
```

File: ci/gpu/build.sh
Comment (jakirkham) [inline ci/gpu/build.sh:77]:
```suggestion
                  "nvidia::cudatoolkit=$CUDA_REL" \
```

**Comment by mike-wendt** [comment]
With rapidsai/gpuci-build-environment#135 and rapidsai/gpuci-tools#12 merged, A100 tests should pass now

**Comment by mike-wendt** [comment]
Incoming fix for the conda builds, waiting on `gpuci-tools` to build again

**Comment by mike-wendt** [comment]
Woot A100 CUDA 11 tests work with all of the above patches. Running a test with a branch build to verify uploads work off of this branch. That combined with these tests can merge this PR

**Comment by mike-wendt** [comment]
Branch build to test upload changes - https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/branches/job/cudf-branch-pipeline/2300/

**Comment by mike-wendt** [comment]
Will merge first thing in the morning after final ops review

**Comment by mike-wendt** [comment]
Successful branch test and two survived 504 errors during upload https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-build/69425/console

May want to retry 3 times instead. Will get feedback before submitting a change

File: ci/cpu/upload.sh
Comment (ajschmidt8) [inline ci/cpu/upload.sh:42]:
```suggestion
gpuci_logger "Starting conda uploads"
```

strikes again 😛

**Review by raydouglass — APPROVED** [review]

File: ci/docs/build.sh
Comment (ajschmidt8) [inline ci/docs/build.sh:32]: This whole `TODO` block can probably be removed. I see `sphinx-markdown-tables` and `beautifulsoup4` in the [rapids-doc-env](https://github.com/rapidsai/integration/blob/branch-0.16/conda/recipes/rapids-doc-env/meta.yaml#L37-L49) and `jq` in our [gpuci devel images](https://github.com/rapidsai/gpuci-build-environment/blob/branch-0.16/rapidsai/devel.Dockerfile#L80-L83).

(GitHub won't let me make this code suggestion since it spans lines of code that you didn't modify)

**Review by ajschmidt8 — COMMENTED** [review]
two small things, otherwise 👍

**Review by dillon-cullinan — APPROVED** [review]

**Comment by mike-wendt** [comment]
Cancelled tests, admin merging due to recent changes changes being a config tweak, doc update, and duplicate code removal

**Pull request merged**
